crates/worker/src/partition/mod.rs

```diff
@@ -531,7 +531,12 @@ where
         self.status.last_applied_log_lsn = Some(lsn);
         self.status.last_record_applied_at = Some(MillisSinceEpoch::now());
         match self.status.replay_status {
-            ReplayStatus::CatchingUp if self.status.target_tail_lsn.is_some_and(|v| lsn >= v) => {
+            ReplayStatus::CatchingUp
+                if self
+                    .status
+                    .target_tail_lsn
+                    .is_some_and(|tail| lsn.next() >= tail) =>
```

Comment:
Why `lsn.next()`? Wouldn't this set us to `Active` when we are processing `tail.prev()`?

Reply:
Yes. the last applied will be the point _before_ the tail

Reply:
At this point, we haven't fully processed this very last record, though. But I guess this solves the problem if there weren't any activity on the log after the `target_tail_lsn`, then we wouldn't move to active, right?

Reply:
Exactly yes.